sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -228,6 +228,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       table: String,
       parts: Array[Partition],
       connectionProperties: Properties): DataFrame = {
+    require(userSpecifiedSchema.isEmpty, "jdbc does not allow user-specified schemas.")
```

Comment:
I know we're in nit territory but "schemas" is fine and "schema" would not be grammatical